### DISCUSSION TITLE
Improve horizontal scrolling for cut off code

### DISCUSSION THREAD

**Discussion by Marcono1234** [community/community#14015] [Pull Requests]
When viewing pull request changes in the unified diff view lines are not wrapped (see also #8497). When lines then overflow the visible area, the GitHub web interface is not handling this very well.

There are two main issues:
1. The horizontal scrollbar only appears at the end of the diff for a file. This makes it impossible to use for files with lots of changes (too large to all fit on the screen vertically), without having to scroll up and down just to access the horizontal scrollbar.
➡️ Expected would be that the horizontal scrollbar is always visible, if necessary floating about the remaining vertically cut off diff of a file.
2. Scrolling horizontally hides the lines numbers. Not being able to see the line numbers is a bit inconvenient, but this has other annoying side effects:
    1. This also hides the buttons for expanding collapsed, unchanged code sections, preventing you from expanding these sections, and requiring you to scroll horizontally just to access these buttons.
    2. You cannot easily select lines, and cannot add review comments without scrolling horizontally to show the line numbers again.

    ➡️ Expected would be that the line numbers are always visible, is necessary floating about the remaining horizontally cut off diff of the line.

Similiar issues also appear in the regular file view of GitHub with long lines.

**Comment by twberger** [comment]
I wanted to report the exact same thing. Screenshot to illustrate the issue.

![image showing that very long code lines can't be seen during PR review](https://user-images.githubusercontent.com/97248410/161228512-0854b937-0352-4710-bac9-4241d8e98374.png)

**Comment by Marcono1234** [comment]
It looks like this might actually be fixed now for the diff code view; lines seem to be wrapped automatically if they would overflow the visible area.
However, for the regular (non-diff) code view this problem still exists (both old and new beta version, see also [comment](https://github.com/orgs/community/discussions/48301#discussioncomment-5111826)).
